serde/src/main/java/com/amazon/ionhiveserde/formats/IonOutputFormat.java

```diff
@@ -76,15 +80,33 @@
                                                              final Progressable progress)
         throws IOException {
 
-        final FileSystem fs = finalOutPath.getFileSystem(jc);
-        final OutputStream out = fs.create(finalOutPath, progress);
-
-        // If we are passed in a reporter, make sure we call incrCounters during write
-        Optional<Reporter> reporter = Optional.empty();
-        if (progress instanceof Reporter) {
-            reporter = Optional.of((Reporter) progress);
+        if (isCompressed) {
+            CompressionCodec codec = getCompressionCodec(jc);
+            Path file = FileOutputFormat.getTaskOutputPath(jc,
```

Comment:
Did you determine whether it was necessary to set a property to make this work?

Reply:
I removed this line and use `finalPath`, which works fine but the output compression file misses extension (e.g. `.gz`).
I am going to do a little more investigation to see if I can add the extension somewhere.